diff --git a/www/sites/all/modules/contrib/wysiwyg_tools_plus/plugins/accordion.js b/www/sites/all/modules/contrib/wysiwyg_tools_plus/plugins/accordion.js
index 0e50fc6..aa233a8 100644
--- a/www/sites/all/modules/contrib/wysiwyg_tools_plus/plugins/accordion.js
+++ b/www/sites/all/modules/contrib/wysiwyg_tools_plus/plugins/accordion.js
@@ -1,7 +1,7 @@
 (function ($) {
 	Drupal.wysiwyg.plugins.accordion = {
 	  invoke: function(data,settings,instanceId) {
-			Drupal.wysiwyg.instances[instanceId].insert('<div class="ready-accordion">&nbsp;' + data.content + '</div>');
+      Drupal.wysiwyg.instances[instanceId].insert('<div class="ready-accordion">' + data.content + '</div>');
 		},
 	}
 }(jQuery));
\ No newline at end of file
